### PR TITLE
Release/0.4.0

### PR DESCRIPTION
**Pull request by HeJones**
### What

- Added script to import region area data into Mongodb, and update script to upsert from insert

### How to review

- Check only changes to the script have been included

### Who can review

- Anyone

### CODE REVIEW

**Review by daiLlew — CHANGES_REQUESTED** [review]
Are there any unit tests that could be added to compliment the new functionality you've added?

File: import-areas/import-areas.go [hunk 1]
```diff
@@ -156,6 +184,125 @@ func postCountryQuery(query string) (*models.Area, error) {
...
 }
 
+func buildAreas(areaStruct AreaStruct) ([]models.Area, error) {
+	ctx := context.Background()
+	areas := make([]models.Area, 0)
+
+	for _, binding := range areaStruct.Results.Bindings {
+		areaType, err := lookupAreaType(binding.AreaCode.Value)
+		logData := log.Data{"AreaCode": binding.AreaCode.Value}
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (HeJones) [reply import-areas/import-areas.go:194]: I have reordered the variable declaration

File: import-areas/import-areas.go [hunk 2]
```diff
@@ -156,6 +184,125 @@ func postCountryQuery(query string) (*models.Area, error) {
...
+			log.Event(ctx, "error returned from areaType lookup", log.ERROR, log.Error(err), logData)
+			return nil, err
+		}
+		area := models.Area{
+			ID:   binding.AreaCode.Value,
+			Name: binding.AreaName.Value,
+			Type: areaType,
+		}
+
+		parentType, _ := lookupAreaType(binding.ParentCode.Value)
```
[the remark this answers is not in the document]
Comment (HeJones) [reply import-areas/import-areas.go:204]: Thank you for spotting this

File: import-areas/import-areas.go [hunk 3]
```diff
@@ -156,6 +184,125 @@ func postCountryQuery(query string) (*models.Area, error) {
...
+		logData = log.Data{"ParentAreaCode": binding.ParentCode.Value}
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (HeJones) [reply import-areas/import-areas.go:206]: thank you for spotting this

File: import-areas/import-areas.go [hunk 4]
```diff
@@ -156,6 +184,125 @@ func postCountryQuery(query string) (*models.Area, error) {
...
+func postAreaQuery(query string) ([]models.Area, error) {
+	ctx := context.Background()
+
+	v := url.Values{}
+	v.Set("query", query)
+
+	client := &http.Client{}
+	req, _ := http.NewRequest("POST", "http://statistics.data.gov.uk/sparql", strings.NewReader(v.Encode()))
+	req.Header.Add("Accept", "application/sparql-results+json")
+	resp, _ := client.Do(req)
```
[the remark this answers is not in the document]
Comment (HeJones) [reply import-areas/import-areas.go:231]: Thank you for spotting this, I have also handled the error on line 229 and for the similar queries in the function above

File: import-areas/import-areas.go [hunk 5]
```diff
@@ -156,6 +184,125 @@ func postCountryQuery(query string) (*models.Area, error) {
...
+
+	data, _ := ioutil.ReadAll(resp.Body)
```
[the remark this answers is not in the document]
Comment (HeJones) [reply import-areas/import-areas.go:233]: I am know handling this error

**Comment by HeJones** [comment]
As this is a script for the moment I have not added any unit tests

**Review by daiLlew — APPROVED** [review]

**Pull request merged**